### ISSUE TITLE
[java] UnnecessaryConstructor FP with annotated ctor

### ISSUE THREAD

**Issue by oowekyala** [pmd/pmd#3791]
**Affects PMD Version:** todo

**Rule:** UnnecessaryConstructor

https://pmd.github.io/latest/pmd_rules_java_codestyle.html#unnecessaryconstructor

**Description:** 

The rule supports an ignored annotation property. We should most likely ignore all annotations by default and remove the property.

https://github.com/pmd/pmd/commit/ee74e7eacf4bdcb221d8cae4927dccd3e33b205f#r66967494

**Code Sample demonstrating the issue:**

```
public class Report {
   @DeprecatedUntil700
    @InternalApi
    public Report() { // NOPMD - UnnecessaryConstructor
    }
}
```

**Expected outcome:**

PMD reports a violation at line ..., but that's wrong. That's a false positive.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
